scripts/idl/test_generators.py

```diff
@@ -119,6 +120,8 @@ def _create_generator(self, storage: GeneratorStorage, idl: Idl):
             return JavaGenerator(storage, idl)
         if self.generator_name.lower() == 'bridge':
             return BridgeGenerator(storage, idl)
+        if self.generator_name.lower() == 'cpp-app':
+            return CppApplicationGenerator(storage, idl)
```

Comment:
We really need a way to tease this out such that the actual generator is hard-coded like this as part of the 'core' IDL compiler + code generator.

Reply:
I do not understand this comment - what would you like to do? Shared dispatch of 'string+storage to generator' so that code is reused between generator and test?

Reply:
I created `idl/generators/registry.py` to centralize the logic of string/generator and construction logic.